addons/sale_timesheet/models/project.py

```diff
@@ -211,9 +211,13 @@ def write(self, values):
             })
         return res
 
+    def _possibly_updated_timesheets(self):
+        self.ensure_one()
+        return self.mapped('timesheet_ids').filtered(lambda t: not t.is_so_line_edited and t._is_not_billed())
+
     def _update_timesheets_sale_line_id(self):
         for project in self.filtered(lambda p: p.allow_billable and p.allow_timesheets):
-            timesheet_ids = project.sudo(False).mapped('timesheet_ids').filtered(lambda t: not t.is_so_line_edited and t._is_not_billed())
+            timesheet_ids = project.sudo(False)._possibly_updated_timesheets()
```

Comment:
I think the `sudo(False)` is useless here (I know you did not add it, Laurent and I added it but I think it was useless. 😇 )
```suggestion
            timesheet_ids = project._possibly_updated_timesheets()
```